spikesorters/utils/shellscript.py

```diff
@@ -53,18 +53,24 @@ def start(self) -> None:
                     script_path = script_path.parent / (script_path.name + '.bat')
                 else:
                     script_path = script_path.parent / (script_path.name + '.sh')
+                script_log_path = script_path.parent / ('spikesorter_log.txt')
         else:
             tempdir = Path(tempfile.mkdtemp(prefix='tmp_shellscript'))
             if 'win' in sys.platform and sys.platform != 'darwin':
                 script_path = tempdir / 'script.bat'
             else:
                 script_path = tempdir / 'script.sh'
+            script_log_path = tempdir / ('spikesorters_log.txt')
             self._dirs_to_remove.append(tempdir)
         self.write(script_path)
         cmd = str(script_path)
         print('RUNNING SHELL SCRIPT: ' + cmd)
         self._start_time = time.time()
-        self._process = subprocess.Popen(cmd)
+        self._process = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.STDOUT,  bufsize = 1, universal_newlines=True)
+        with open(script_log_path, 'w+') as script_log_file:
+            for line in self._process.stdout:
+                script_log_file.write(line)
+                print(line)
```

Comment:
This print must controlled by a "verbose" propagated from the the verbose of sorter class. No ?

Reply:
By this, you mean that the console/jupyter notebook should display the run time trace only if a  certain option (say "verbose") is set to be true?

Reply:
Yes, the base sorter has this option: https://github.com/SpikeInterface/spikesorters/blob/master/spikesorters/basesorter.py#L45

If verbose is False, I think the output shoudl be printed to the runtime trace only.

Is it possible to do that?

Reply:
It can be done if I pass the 'verbose' property into the shellScript object that we create in the sorter subclass implementation. Will that be okay?

Reply:
Yes of course :) Thanks again! 

BTW, can you call the `getConsoleLog` --> `get_console_log`?

Reply:
Yeah, definitely! :)

Reply:
https://github.com/manimoh/spikesorters/blob/4dfefe267e311cc33884b8d6e3b612be47f1c93b/spikesorters/utils/shellscript.py#L73-L74